lib/shoryuken/manager.rb

```diff
@@ -13,16 +13,18 @@ def initialize(fetcher, polling_strategy)
       @queues = Shoryuken.queues.dup.uniq
 
       @done = Concurrent::AtomicBoolean.new(false)
-      @dispatching = Concurrent::AtomicBoolean.new(false)
 
       @fetcher = fetcher
       @polling_strategy = polling_strategy
 
-      @heartbeat = Concurrent::TimerTask.new(run_now: true,
-                                             execution_interval: HEARTBEAT_INTERVAL,
-                                             timeout_interval: 60) { dispatch }
+      Concurrent::TimerTask.new(execution_interval: 1) do
+        Shoryuken.logger.info "Threads: #{Thread.list.size}"
+      end.execute
 
       @pool = Concurrent::FixedThreadPool.new(@count, max_queue: @count)
+      @dispatcher_pool = Concurrent::SingleThreadExecutor.new
```

Comment:
Except that `TimerTask` is the thing that's leaking threads because of the ones it creates to do the timeout. But I'm sure the execution task occurs faster than the heartbeat period.

Don't really need a `max_queue` in `TimerTask` because the way the class is designed it should be locking around the actual task and only letting one operate at a time. 

But the lack of a correct implementation for the timeout causes thread leaks in the timeout monitor if the task takes longer than the execution interval (it should be the timeout interval at the very least, but it never refers to this… hence the bug) to complete. 

The current design intention of `TimerTask` seems to have it limited to only ever using 2 threads.